django/db/models/expressions.py

```diff
@@ -490,6 +490,27 @@ def desc(self):
         return OrderBy(self, descending=True)
 
 
+class ResolvedOuterRef(F):
+    """
+    An object that contains a reference to an outer query.
+
+    In this case, the reference to the outer query has been resolved, because the
+    inner query has been used as a subquery.
+    """
+    def as_sql(self, *args, **kwargs):
+        raise ValueError('This queryset contains a reference to an outer query, and cannot be '
+                         "evaluated in its own right")
```

Comment:
I can't think of a better message now, nor do I have the inclination to find a better exception. This is more just cleanup, to show the intent.